### ISSUE TITLE
Create "Advanced Options" in Replication Settings

### ISSUE THREAD

**Issue by nataliekwong** [airbytehq/airbyte#11850]
There are a few replication settings that are only sparingly used by power users:

- Connection data residency
- Stream Prefix

Suggestion: do not expand these immediately, but give the option to edit these before stream setup. 

---
We can hide the connection settings during the connection setup, so that users aren't inhibited by the last step and can more quickly get the first connection setup. These settings can still be available to the user, but hidden by a toggle or dropdown where the user can access them if they want to change any of the defaults.

Example mockup: https://www.figma.com/file/etaOkOsnXMoKwbfhCD71kw/Web-app?node-id=28072%3A100930

**Comment by octavia-squidington-iii** [comment]
cc @airbytehq/frontend

**Comment by nataliekwong** [comment]
Blocked by https://github.com/airbytehq/airbyte/issues/11849 (we need to know whether the table selection should be collapsed or not, and knowing how often users are syncing all or some streams will determine this aspect.

**Comment by timroes** [comment]
Adding https://rmoff.net/2022/11/08/data-engineering-in-2022-elt-tools/ as a reference. This article lists a lot of nice examples how presenting all those options is confusing initially, and why it would be good to have a "minimal" form presented initially and have the rest being "advanced settings".

**Comment by nataliekwong** [comment]
@timroes I believe this is addressed by the revision of the connection settings page currently launched behind a feature flag. Is it fair to close this?

**Comment by timroes** [comment]
I don't think we have any work done on this issue so far, so I'd like to keep this open, since I still think this is a valuable addition.

**Comment by nataliekwong** [comment]
We do have a new minimized version of the connection settings (not for setup, but for existing connections) which was the original intent of the issue:
![Screenshot 2023-03-07 at 5 01 29 PM](https://user-images.githubusercontent.com/38087517/223592057-373309c0-72d6-43d6-94ed-d984374ebe37.png)

Given we have received feedback about the namespace setting being unclear (that this makes the first sync experience poor), I would be hesitant to completely minimize the settings as in the view above. This new form is already a slight improvement to the existing setup in the way it groups the various settings.
![Screenshot 2023-03-07 at 5 13 30 PM](https://user-images.githubusercontent.com/38087517/223593792-9aeddad1-c38a-45fb-a15a-19cd4b0e4457.png)

The article does also mention the confusion around selection of the cursor field and the primary key, which shouldn't be an issue any longer given we now automatically set the Sync Mode to Full refresh | Overwrite when the primary key is missing.  One way to take this even further would be to grey out this area for first connections and add a "configure" button to unlock the editing of the form. 

Outside of the configuration settings, seems like we would definitely want to:
- Create Advanced Settings
- Group "Data Residency", "Non-breaking schema updates detected", "Destination Stream Prefix" into Advanced Settings

**Comment by nataliekwong** [comment]
Closing in favor of https://github.com/airbytehq/airbyte/issues/29896 now that we have a final design there that incorporates the creation of 'advanced' settings.
